htdocs/adherents/stats/geo.php

```diff
@@ -90,14 +90,14 @@
         $tab = 'statsstate';
 
         $data = array();
-        $sql .= "SELECT COUNT(d.rowid) as nb, MAX(d.datevalid) as lastdate, MAX(s.dateadh) as lastsubscriptiondate, co.code, co.label, c.nom as label2"; //
+        $sql .= "SELECT COUNT(DISTINCT d.rowid) as nb, MAX(d.datevalid) as lastdate, MAX(s.dateadh) as lastsubscriptiondate, co.code, co.label, c.nom as label2"; //
         $sql .= " FROM ".MAIN_DB_PREFIX."adherent as d";
         $sql .= " LEFT JOIN ".MAIN_DB_PREFIX."c_departements as c on d.state_id = c.rowid";
         $sql .= " LEFT JOIN ".MAIN_DB_PREFIX."c_regions as r on c.fk_region = r.code_region";
         $sql .= " LEFT JOIN ".MAIN_DB_PREFIX."c_country as co on d.country = co.rowid";
         $sql .= " LEFT JOIN ".MAIN_DB_PREFIX."subscription as s ON s.fk_adherent = d.rowid";
         $sql .= " WHERE d.entity IN (".getEntity('adherent').")";
-        $sql .= " AND d.statut = 1";
```

Comment:
By definition a "draft" object in dolibarr is an object that does not exist (it is a draft that may be deleted or validated to exists officially), so we should exclude them from statistitcs.

Reply:
OK, but for historical stats, and comparison, you should include the 
retired adherent, maybe de van check if there's a validation date 
(d.datevalid != "0000-00-00 00:00:00") ?
Otherwise it is a wrong info.

Reply:
Yes filtering on -1 is better. And yes, including the terminated member for stats is usefull. But it also break a feature used at end of year by association to know how many valid/active member there is (required for the annualy moral report).
So the only solution i see is to introduce a filter with 3 choices:
Empy choice
Open only
Terminated only

Filter empty -> All members (except draft). What you expect
Filter on open only status = 1 (like today). So we don't loose a feature.
Filter on terminated status = 0. New feature

It is more work, but the only solution i see to provide best of the 2 needs (current and the one you are looking for).